modal/_serialization.py

```diff
@@ -15,11 +17,11 @@ def __init__(self, buf):
         super().__init__(buf, protocol=PICKLE_PROTOCOL)
 
     def persistent_id(self, obj):
-        if not isinstance(obj, _Handle):
+        if not isinstance(obj, (_Handle, Handle, AioHandle)):
             return
         if not obj.object_id:
             raise InvalidError(f"Can't serialize object {obj} which hasn't been created.")
-        return obj.object_id
+        return (obj.object_id, getattr(obj, TARGET_INTERFACE_ATTR, None), obj._handle_proto())
```

Comment:
maybe put `getattr(obj, TARGET_INTERFACE_ATTR, None)` in a utility function? it took me a while to realize what it does

Reply:
That utility function might belong in synchronicity though right? I'll add one here in the meanwhile to make the intent clear